File: python/proverb/proverb.py
def proverb(*args, qualifier=None):
    res = []
    if len(args) == 0:
        return res
    for num in range(int(len(args) - 1)):
        res.append(f"For want of a {args[num]} the {args[num + 1]} was lost.")
    if qualifier:
        res.append(f"And all for the want of a {qualifier} nail.")
    else:
        res.append(f"And all for the want of a {args[0]}.")
    return res

File: csharp/proverb/Proverb.cs
using System.Linq;

public static class Proverb
{
    public static string[] Recite(string[] subjects)
        => subjects.Length == 0
            ? subjects
            : Enumerable.Range(0, subjects.Length - 1)
                .Select(i => string.Format($"For want of a {subjects[i]} the {subjects[i + 1]} was lost."))
                .Concat(new[] { $"And all for the want of a {subjects[0]}." }).ToArray();
}
